src/DearImGui.hs

```diff
@@ -404,6 +405,20 @@ arrowButton strId dir = liftIO do
     Raw.arrowButton strIdPtr dir
 
 
+-- | Image Area to draw a texture
+--
+-- Calls Raw.image with Default arguments.
+-- See https://github.com/ocornut/imgui/wiki/Image-Loading-and-Displaying-Examples#About-texture-coordinates for an example
+image :: MonadIO m => Ptr() -> ImVec2 -> m()
```

Comment:
This `Ptr ()` can be an opaque handle given out by the opengl backend:

```haskell
thisOne <- OpenGL3.create2DTexture 256 256
...
image thisOne (IMVec2 100 100)
```

Reply:
Hm... I'm not sure if it is this trivial:
If we would add an opaque ptr and a backend to create the textures we would also have to provide all the functions to manage the textures (update, delete). The related Texture-functions are:
`glDeleteTextures, glGenTextures, glGet, glGetTexParameter, glIsTexture, glTexImage1D, glTexImage2D, glTexImage2DMultisample, glTexImage3D, glTexImage3DMultisample, glTexBuffer, glTexParameter`
And this is just for OpenGL - do we really want to wrap all of them?

For example [SDL](https://github.com/libsdl-org/SDL/blob/3c028141946aa9466f8c21fad2546a1cc2467932/src/render/SDL_sysrender.h#L36) is managing this and the wrapper is creating the [Texture](https://hackage.haskell.org/package/sdl2-2.5.3.0/docs/SDL-Video-Renderer.html#t:Texture)-Type. But this holds a lot of meta-information.
Another way would be to use the SDL-mechanisms - but this kind of abstraction is not provided by FLGW.

For OpenGL textures are represented as GLuint. For Vulkan I'm not sure, but I guess it is [Vulkan.Extensions.Handles.Device Memory](https://hackage.haskell.org/package/vulkan-3.11.4/docs/src/Vulkan.Core10.Handles.html#DeviceMemory).

At the moment I have no good idea - main problem is that textures can be deleted in the memory and still be accessable - so all of these operations should have some unsafe-prefix if not managed by some backend.

Reply:
> main problem is that textures can be deleted in the memory and still be accessable - so all of these operations should have some unsafe-prefix if not managed by some backend.

That's nasty indeed.

Perhaps it is too early to provide a wrapper for images. Raw function call is okay and can be provided right away.

Having only the raw call would signal some wrapping is in order, but it is engine- and backend-dependent. That much can be stated in its description.